core/src/main/java/io/github/tcdl/msb/adapters/mock/MockAdapter.java

```diff
@@ -46,27 +55,30 @@ public void publish(String jsonMessage) {
 
     @Override
     public void subscribe(RawMessageHandler messageHandler) {
-
-        Thread subscriberThread = new Thread(() -> {
-            String jsonMessage = null;
-            while (true) {
-                jsonMessage = pollJsonMessageForTopic(topic);
-
-                if (messageHandler != null && jsonMessage != null) {
-                    LOG.debug("Process message for topic {} [{}]", topic, jsonMessage);
-                    messageHandler.onMessage(jsonMessage);
-                } else {
-                    try {
-                        Thread.sleep(CONSUMING_INTERVAL);
-                    } catch (Exception e) {
-                        LOG.debug("Finish listen for subscribed topic");
+        if (executorService == null) {
+            LOG.warn("Mock Adapter not initialized for consuming");
+        } else {
+            executorService.execute(() -> {
+                {
```

Comment:
This is dangerous because it's an infinitely running task. And ExecutorService may starve if we're not careful